dvc/repo/metrics/show.py

```diff
@@ -66,13 +66,42 @@ def _read_typed_metric(typ, xpath, fd):
     return ret
 
 
+def _format_csv(content, delimiter):
+    import csv
+
+    reader = csv.reader(content.split("\n"), delimiter=builtin_str(delimiter))
+    rows = [row for row in reader]
+    columns = zip(*rows)
+    max_widths = [max(map(len, column)) for column in columns]
+    lines = [
+        " ".join(
+            "{entry:{width}}".format(entry=entry, width=width + 2)
+            for entry, width in zip(row, max_widths)
+        )
+        for row in rows
+    ]
+
+    return "\n".join(lines)
+
+
+def _format_output(content, typ):
+    if "\n" not in content or not typ:
```

Comment:
This kind of checks are not safe for CSV that might contain entries with newlines. Could we try to get rid of the `"\n" not in content` condition and handle it with a regular `_format_csv`, even if it's a single line

Reply:
:+1: